File: backend/app/models/calendar.py
"""Calendar event models."""

from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from app.core.database import Base


class CalendarEvent(Base):
    """Calendar event model."""
    
    __tablename__ = "calendar_events"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(255), nullable=False)
    start_time = Column(DateTime, nullable=False)
    end_time = Column(DateTime, nullable=False)
    description = Column(Text, nullable=True)
    attendees = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    
    user = relationship("User", back_populates="calendar_events")

File: frontend/src/types/calendar.ts
export interface CalendarEvent {
  id: number
  user_id: number
  title: string
  description?: string
  start_time: string
  end_time: string
  created_at: string
}

export interface CalendarEventCreate {
  title: string
  description?: string
  start_time: string
  end_time: string
}
